src/com/pty4j/unix/UnixPtyProcess.java

```diff
@@ -230,6 +232,26 @@ private void execInPty(String[] command, String[] environment, String workingDir
           Thread.currentThread().interrupt();
         }
       }
+
+      boolean init = Boolean.getBoolean("unix.pty.init");
+      if (init) {
+        int cols = Integer.getInteger("unix.pty.cols", 80);
+        int rows = Integer.getInteger("unix.pty.rows", 25);
+        WinSize size = new WinSize(cols, rows);
+
+        // On OSX, there is a race condition with pty initialization
+        // If we call com.pty4j.unix.Pty.setTerminalSize(com.pty4j.WinSize) too early, we can get ENOTTY
+        boolean retry = true;
+        for (int attempt = 0; attempt < 1000 && retry; attempt++) {
```

Comment:
Can it hang for too long if it constantly gives ENOTTY?

Reply:
Once the terminal initialized, we can perform `myPty.setTerminalSize(size);` without exceptions.
If the terminal initialization logic hangs, we have a bigger problem.